### PR TITLE
[MAINTENANCE] Run marker tests on drafts

### PR DESCRIPTION
**Pull request by Kilo59**
Turns on marker tests for draft PR's.
This should prevent edge cases where a developer opens a PR as a draft and then converts to regular PR while making no further commits.

### Questions
Is the `if: github.event.pull_request.draft == false` logic really the cause of the PR 👇 making past our CI tests when it should have failed? 

- https://github.com/great-expectations/great_expectations/pull/8562

The sqlite test is a "required" status check so presumably it wouldn't matter if it failed or was skipped, merge should be blocked if it doesn't pass.

----------

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 [![Python Versions](https://img.shields.io/pypi/pyversions/great_expectations.svg)](https://pypi.python.org/pypi/great_expectations)
 [![PyPI](https://img.shields.io/pypi/v/great_expectations)](https://pypi.org/project/great-expectations/#history)
 [![PyPI Downloads](https://img.shields.io/pypi/dm/great-expectations)](https://pypistats.org/packages/great-expectations)
-[![Build Status](https://img.shields.io/azure-devops/build/great-expectations/bedaf2c2-4c4a-4b37-87b0-3877190e71f5/1)](https://dev.azure.com/great-expectations/great_expectations/_build/latest?definitionId=1&branchName=develop)
+[![ci](https://github.com/great-expectations/great_expectations/actions/workflows/ci.yml/badge.svg?event=schedule)](https://github.com/great-expectations/great_expectations/actions/workflows/ci.yml)
```
Comment (Kilo59) [inline README.md:5]: Replacing this old broken Azure CI badge with one that shows the status of our scheduled Github Action run

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM!

**Review by cdkini — APPROVED** [review]

**Review by anthonyburdi — APPROVED** [review]

**Comment by Kilo59** [comment]
No longer needed now that we are using merge queues
